geometric_kernels/lab_extras/extras.py

```diff
@@ -52,3 +52,27 @@ def sinh(x: B.Numeric) -> B.Numeric:
     .. math:: cosh(x) = \frac{\exp(x) - \exp(-x)}{2} ,
     """
     return 0.5 * (B.exp(x) - B.exp(-x))
+
+
+@dispatch
+@abstract()
+def degree(a):
+    """
+    Diagonal matrix with x as main diagonal.
+    """
+
+
+@dispatch
+@abstract()
+def eigenpairs(L, k: int):
+    """
+    Obtain the k highest eigenpairs of a symmetric PSD matrix L.
+    """
+
+
+@dispatch
+@abstract()
+def set_value(a, index: B.Numeric, value: B.Numeric):
```

Comment:
is this an inplace or copy operation? The behaviour between the different backends doesn't seem consistent. Would it be possible to make it consistent? Probably making it *not* in-place is the safest thing to do.

Reply:
I think the only way to make it consistent is to do it not in-place as tf and jax arrays are immutable. I can make this change.

Reply:
Resolved.